File: post-render.py
import os
import shutil

outputs = os.getenv("QUARTO_PROJECT_OUTPUT_FILES").split("\n")

for output in outputs:
  if not any("index_files" in line for line in open(output)):
    shutil.rmtree(os.path.join(os.path.dirname(output), "index_files"), ignore_errors = True)

File: post-render.R
outputs <- strsplit(Sys.getenv("QUARTO_PROJECT_OUTPUT_FILES"), "\n", fixed = TRUE)[[1]]

for (output in outputs) {
  if (!any(grepl("index_files", readLines(output, warn = FALSE), fixed = TRUE))) {
    unlink(file.path(dirname(output), "index_files"), recursive = TRUE)
  }
}
